### ISSUE TITLE
2019/06/19/cqrs-symfony-elasticsearch-react

### ISSUE THREAD

**Issue by romaricdrigon** [romaricdrigon/romaricdrigon.github.io#5]
Please comment below!
Comments will automatically be published on the blog, too.

**Comment by Gompali** [comment]
Hi, interesting article, pointing out implementation of CQRS. How the "write" database can publish to the "read" database ? Do you have an example ?

**Comment by romaricdrigon** [comment]
You want to use a bus, and technically a message queue makes more sense: having an asynchronous transport allows to lighten the load on the write database.
I don't have personal examples, but you find plenty around. Someone implemented it [with Tactician](https://github.com/phpgears/cqrs-tactician), and I'm sure you could do the same with Symfony Messenger.

**Comment by Gompali** [comment]
Yes, I'm ok with messenger. In terms of design, am I right that :

CommandBus -> Create entity in SQL
EventBus -> Sends "EntityCreated" to event handler that should feed let say a MongoDB 
QueryBus -> Sends queries to MongoDB

I'm looking for event handling patterns to implement properly and without vendors ;-)

**Comment by romaricdrigon** [comment]
It sounds correct, theory-wise.
In practice, I'm not sure you really need 3 buses. I would use Doctrine listeners instead of a CommandBus in most cases, and QueryBus is not quite something "physically". You just query MongoDB.
